src/hmac.rs

```diff
@@ -123,13 +115,44 @@ pub type Signature = Tag;
 #[derive(Clone, Copy, Debug)]
 pub struct Tag(digest::Digest);
 
+impl Tag {
+    /// Do a constant time equality check that is expected to succeed.
+    ///
+    /// The result of the comparison trait `PartialEq` is always a boolean. Verifying a signature
+    /// should however result in a `Result` for ease of use.
+    fn check<T: ?Sized>(&self, rhs: &T) -> Result<(), error::Unspecified>
+        where Self: PartialEq<T>
+    {
+        match self == rhs {
+            true => Ok(()),
+            false => Err(error::Unspecified),
+        }
+    }
+}
+
 impl AsRef<[u8]> for Tag {
     #[inline]
     fn as_ref(&self) -> &[u8] {
         self.0.as_ref()
     }
 }
 
+impl PartialEq for Tag {
```

Comment:
Good points, then I'm for removing these altogether and replace the trait bound on `verify_parts` with `AsRef<[u8]>`. Since we are discussing this, might `Borrow<[u8]>` be a slight better fit anyways? For `AsRef` also includes deeper and more surprising implementations as `String: AsRef<[u8]>`.

Reply:
> For AsRef also includes deeper and more surprising implementations as String: AsRef<[u8]>.

Good catch, I didn't think of this. I think it's another reason to prefer `tag: &[u8]` everywhere.